tachyon_core/src/execution/node/aggregate.rs

```diff
@@ -11,110 +13,201 @@ pub enum AggregateType {
     Average,
 }
 
+struct AggregateChild {
+    node: Box<TNode>,
+    /// Next vector is stored here if looked at but not returned
+    peeked_vector: Option<Vector>,
+    /// End timestamp of aggregation (sub)period
+    end: Timestamp,
+    /// Set to true when all vectors have been read
+    done: bool,
+}
+
 pub struct AggregateNode {
     pub aggregate_type: AggregateType,
-    child: Box<TNode>,
-    other_child: Option<Box<TNode>>,
-    returned: bool, // Used to distinguish between returning 0 and None when summing/counting no values
+    subperiod: Option<Duration>,
+    end: Timestamp,
+    child: AggregateChild,
+    other_child: Option<AggregateChild>,
 }
 
 impl AggregateNode {
     pub fn new(
         aggregate_type: AggregateType,
+        subperiod: Option<Duration>,
+        start: Timestamp,
+        end: Timestamp,
         child: Box<TNode>,
         other_child: Option<Box<TNode>>,
     ) -> Self {
+        let curr_end = if let Some(subperiod) = subperiod {
+            start + (subperiod.as_millis() as u64)
```

Comment:
With this, does this now imply that all timestamps are now milliseconds? I don't think we made this distinction before.

I think this is fine, but we just need to communicate this to our users. Also, this means we can display dates better in CLI maybe?